src/main/java/com/rackspace/feeds/feedscatalog/ResolveHostFilter.java

```diff
@@ -0,0 +1,94 @@
+package com.rackspace.feeds.feedscatalog;
+
+import com.rackspace.feeds.filter.TransformerUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+
+public class ResolveHostFilter implements Filter {
+
+    static Logger LOG = LoggerFactory.getLogger(ResolveHostFilter.class);
+    static final String resolveHostXslt = "feedscatalog-resolve-host.xsl";
+
+    private TransformerUtils transformer;
+
+    /**
+     * This method is called once when the filter is first loaded.
+     */
+    public void init(FilterConfig filterConfig) throws ServletException {
+        try {
+            URL xsltResource = Thread.currentThread().getContextClassLoader().getResource(resolveHostXslt);
+            if (xsltResource != null) {
+                transformer = TransformerUtils.getInstanceForXsltAsFile(xsltResource.getPath());
+            }
+            else {
+                throw new UnavailableException("resource is null - " + resolveHostXslt);
+            }
+        } catch ( Exception e ) {
+            LOG.error( "Error loading xslt: " + e.getMessage());
+            throw new ServletException( e );
+        }
+    }
+
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        HashMap<String, Object> params = new HashMap<String, Object>();
+        putTenantIdParams(httpServletRequest, params);
+        putExternalLocParams(httpServletRequest, params);
+
+        transformer.doTransform(httpServletRequest, httpServletResponse, httpServletResponse, chain, params);
+    }
+
+    /**
+     * The counterpart to the init( ) method.
+     */
+    public void destroy( ) {
+
+    }
+
+    void putTenantIdParams(HttpServletRequest request, HashMap<String, Object> params) {
+        //put tenantId and nastId into the params HashMap
+        //tenantId is the shortest length value, nastId is the longest length value
+        Enumeration headerValues = request.getHeaders("x-tenant-id");
+        ArrayList<String> tenantIds = new ArrayList<String>();
+        while (headerValues.hasMoreElements()) {
+            tenantIds.add((String) headerValues.nextElement());
+        }
+
+        if (tenantIds.size() > 0) {
```

Comment:
If there are no tenantIds in the header, the catalog simply returns with the href without the tenantId